src/compiler/scala/tools/nsc/tasty/TreeUnpickler.scala

```diff
@@ -0,0 +1,1115 @@
+          case OPAQUE => addFlag(Opaque)
+          case STATIC => addFlag(Static)
+          case OBJECT => addFlag(Object)
+          case TRAIT => addFlag(Trait)
+          case ENUM => addFlag(Enum)
+          case LOCAL => addFlag(Local)
+          case SYNTHETIC => addFlag(Synthetic)
+          case ARTIFACT => addFlag(Artifact)
+          case MUTABLE => addFlag(Mutable)
+          case FIELDaccessor => addFlag(FieldAccessor)
+          case CASEaccessor => addFlag(CaseAccessor)
+          case COVARIANT => addFlag(Covariant)
+          case CONTRAVARIANT => addFlag(Contravariant)
+          case SCALA2X => addFlag(Scala2x)
+          case DEFAULTparameterized => addFlag(DefaultParameterized)
+          case STABLE => addFlag(Stable)
+          case EXTENSION => addFlag(Extension)
+          case GIVEN => addFlag(Implicit)
+          case PARAMsetter => addFlag(ParamSetter)
+          case PARAMalias => addFlag(ParamAlias)
+          case EXPORTED => addFlag(Exported)
+          case OPEN => addFlag(Open)
+          case PRIVATEqualified =>
+            readByte()
+            privateWithin = readWithin(ctx)
+          case PROTECTEDqualified =>
+            addFlag(Protected)
+            privateWithin = readWithin(ctx)
+          case ANNOTATION =>
+            annotFns = readAnnot(ctx) :: annotFns
+          case tag =>
+            assert(assertion = false, s"illegal modifier tag ${astTagToString(tag)} at $currentAddr, end = $end")
+        }
+      }
+      (flags, if (ctx.ignoreAnnotations) Nil else annotFns.reverse, privateWithin)
+    }
+
+    private val readTypedWithin: Context => Symbol = implicit ctx => readType().typeSymbolDirect
+
+    private val readTypedAnnot: Context => DeferredAnnotation = { implicit ctx =>
+      val annotCtx = ctx.addMode(ReadAnnotation)
+      val start = currentAddr
+      ctx.log(s"<<< $start reading annotation:")
+      readByte() // tag
+      val end      = readEnd()
+      val annotSym = readType()(annotCtx).typeSymbolDirect
+      val deferred = readLaterWithOwner(end, rdr => ctx => {
+        ctx.log(s"${rdr.reader.currentAddr} reading LazyAnnotationRef[${annotSym.fullName}](<lazy>)")
+        rdr.readTerm()(ctx)
+      })(annotCtx)
+      ctx.log(s">>> $start LazyAnnotationRef[${annotSym.fullName}](<lazy>)")
+      new DeferredAnnotation(deferred)
+    }
+
+    /** Create symbols for the definitions in the statement sequence between
+     *  current address and `end`.
+     */
+    def indexStats(end: Addr)(implicit ctx: Context): Unit = {
+      while (currentAddr.index < end.index) {
+        nextByte match {
+          case tag @ (VALDEF | DEFDEF | TYPEDEF | TYPEPARAM | PARAM) =>
+            symbolAtCurrent()
+            skipTree()
+          case IMPORT =>
+            skipTree()
+          case PACKAGE =>
+            processPackage(end => implicit ctx => indexStats(end))
+          case _ =>
+            skipTree()
+        }
+      }
+      ctx.forceAnnotations()
```

Comment:
in the present implementation this actually does not deal with the case where a lookahead symbol definition also declares annotations, e.g.
```scala
object DelayedChained {

  sealed trait Root

  object Internal {

    sealed trait Branch extends Root

    object Deeper {
      final class LeafA extends Branch
      final class LeafB extends Branch
    }

  }

}

```

Reply:
new commit adds so that when we lookahead to create a symbol we also force annotations collected there.